sbncode/Calibration/TrackCaloSkimmer_module.cc

```diff
@@ -129,6 +132,22 @@ void sbn::TrackCaloSkimmer::analyze(art::Event const& e)
   auto const dprop =
     art::ServiceHandle<detinfo::DetectorPropertiesService const>()->DataFor(e, clock_data);
 
+  // Identify which detector: can only detect either sbnd or icarus
+
+  std::string gdml = std::filesystem::path(geometry->GDMLFile()).filename();
+  for(unsigned int i = 0; i <gdml.size(); ++i) gdml[i] = std::tolower(gdml[i]); 
+
+  const bool hasSBND = ((gdml.find("sbnd") != std::string::npos) ||
+			(geometry->DetectorName().find("sbnd") != std::string::npos));
+
+  const bool hasIcarus = ((gdml.find("icarus") != std::string::npos) ||
+			(geometry->DetectorName().find("icarus") != std::string::npos));
+
+  if(hasSBND == hasIcarus) { 
+    std::cout << "Unable to automatically determine either SBND or ICARUS!" << std::endl;
```

Comment:
Can we put a little more context in this message so people can figure out what happened if this goes wrong? eg start it with `TrackCaloSkimmer: `